parts/linux/cloud-init/artifacts/cse_install.sh

```diff
@@ -200,102 +200,122 @@ downloadSecureTLSBootstrapKubeletExecPlugin() {
     fi
 }
 
-downloadContainerdWasmShims() {
-    declare -a wasmShimPids=()
-    local containerd_wasm_filepath="/usr/local/bin"
-    BOOTSTRAP_PROFILE_CONTAINER_REGISTRY_SERVER="${BOOTSTRAP_PROFILE_CONTAINER_REGISTRY_SERVER:=}"
-    
-    for shim_version in $CONTAINERD_WASM_VERSIONS; do
-        binary_version="$(echo "${shim_version}" | tr . -)"
-
-        local version_suffix
-        local shims_to_download=()
-        local shim_prefix="containerd-shim-"
-        local registry_path
-        local base_path
-        local shim_filename
-
-        # figure out version suffix, shims to download, and paths
-        if [ "$shim_version" == "v0.15.1" ]; then
-            version_suffix="-v2"
-            shims_to_download=("spin")
-            registry_path="oss/binaries/spinkube/containerd-shim-spin"
-            base_path="spinkube"
-            shim_filename="containerd-shim-spin-v2"
-        else
-            version_suffix="-v1"
-            shims_to_download=("spin" "slight")
-            registry_path="oss/binaries/deislabs/containerd-wasm-shims"
-            base_path="containerd-wasm-shims"
-            shim_filename="containerd-wasm-shims-linux-${CPU_ARCH}.tar.gz"
-            if [ "$shim_version" == "v0.8.0" ]; then
-                shims_to_download+=("wws")
-            fi
-        fi
+# Install, download, update wasm must all be run from the same function call
+# in order to ensure WASMSHIMPIDS persists correctly since in bash a new
+# function call from install-dependnecies will create a new shell process.
+installContainerdWasmShims(){
+    local download_location=${1}
+    PACKAGE_DOWNLOAD_URL=${2}
+    shift 2 # shift past the first 2 arguments to capture the list of versions
+    local package_versions=("$@")
+
+    local shims_to_download=("spin" "slight")
+    local version_suffix="-v1"
+    local mcr_registry_path="deislabs/containerd-wasm-shims"
+    local shim_filename="containerd-wasm-shims-linux-${CPU_ARCH}.tar.gz"
+    if [ "$shim_version" == "0.15.1" ]; then
+        version_suffix="-v2"
+        shims_to_download=("spin")
+        mcr_registry_path="spinkube/containerd-shim-spin"
+        shim_filename="containerd-shim-spin-v2"
+    elif [ "$shim_version" == "0.8.0" ]; then
+        shims_to_download+=("wws")
+    fi
```

Comment:
suggest: Given we have made two separate components for containerd-wasm-shims and spinkube, I'd suggest we have two separate `install` functions.

1. `installContainerdWasmShims()` installs the `slight, spin, wws` shims before version `v0.15.1`
2. `installSpinKube()` installs the `spin` shim with version `v0.15.1`